x-pack/plugin/async-search/qa/rest/src/test/resources/rest-api-spec/test/async-search/10_deprecation.yml

```diff
@@ -0,0 +1,99 @@
+---
+"Async search":
+  - skip:
+      features: "warnings"
+
+  - do:
+      indices.create:
+        index: test-1
+        body:
+          settings:
+            number_of_shards: "2"
+
+  - do:
+      indices.create:
+        index: test-2
+        body:
+          settings:
+            number_of_shards: "1"
+
+  - do:
+      indices.create:
+        index: test-3
+        body:
+          settings:
+            number_of_shards: "3"
+
+  - do:
+      index:
+        index:  test-2
+        body:   { max: 2 }
+
+  - do:
+      index:
+        index:  test-1
+        body:   { max: 1 }
+
+  - do:
+      index:
+        index:  test-3
+        body:   { max: 3 }
+
+  - do:
+      indices.refresh: {}
+
+  - do:
+      warnings:
+        - '[deprecated] query'
+      async_search.submit:
+        index: test-*
+        wait_for_completion_timeout: 10s
+        body:
+          query:
+            deprecated: {}
+
+  - is_false: id
+  - match:  { is_partial:                   false }
+  - length: { response.hits.hits:               3 }
+
+  - do:
+      warnings:
+        - '[deprecated] query'
+      async_search.submit:
+        index: test-*
+        wait_for_completion_timeout: 10s
+        keep_on_completion: true
+        body:
+          query:
+            deprecated: {}
+
+  - set:    { id:                              id }
+  - match:  { is_partial:                   false }
+  - length: { response.hits.hits:               3 }
```

Comment:
nit: is it necessary to have both these two submit? if so maybe they should be part of a different test section (one only for submit, the other one to test get and delete)

Reply:
We test the case where the response is stored (`keep_on_completion` set to true) so I think it's necessary. We also need to record the search id that is returned by the submit so they should be part of the same section ?

Reply:
I mean that the previous submit is tested on its own, while the second one and the following get and delete depend on each other, did I get that right?

Reply:
Ok got it, I can split the two.